### DISCUSSION TITLE
Official integration

### DISCUSSION THREAD

**Discussion by simieski** [Hyundai-Kia-Connect/kia_uvo#249] [General]
I've been using this custom component for about 9 months since I got my new KIA eNiro. Thanks so much for all your hard work. I've been mighty impressed at how rock solid it is. Asking Alexa or Siri to warm up the car has been a god send this winter. I've even made a actionable notification using HA iOS app, where if the temp has been freezing overnight I'm asked at 8am if I'd like to 'warm the car'. I would share, but I need to update for the upcoming changes to actions first. 

Keep up the good work!! A few things:
- Let me know if I can help with testing anything specific to eNiro
- Have you ever considered submitting this as an official KIA integration to Home Assistant? 
- I was glad to see you've added some options to change the update interval, do you have any inkling on what the rate limit might be with their API? As I'd like to test out some automations which require the most up to date information (eg. 1 minute).

**Comment by cdnninja** [comment]
Thanks for the support!  It has been a team effort!    Very cool to hear how people are using this integration in day to day life.   

What region are you in?   The biggest thing we need testing on is our new API. We have essentially refactored this whole thing for a 2.0 that supports multi car but also follows more of what is needed to be a core integration.    We have an open pull request with the HA core group but it is still going through review https://github.com/home-assistant/core/pull/62451.  The "core" code is also in draft here for testing: https://github.com/fuatakgun/kia_uvo/issues/219

How I see this going is that branch once close to feature parity will be merged to turn this to 2.0  This will allow us testing and to migrate to the state that the core migration will be in.   Hopefully providing us testing and a migration path.

**Reply by simieski** [reply]
Great. Sounds like great progress. I don’t have multi car. I’m in UK (Europe)
Let me know if you want me to test API version 2. I work as a product manager in app development so I’m familiar with bug reporting processes.

**Reply by cdnninja** [reply]
It would be appreciated! As mentioned in the #219 link much is missing but not a bad thing for us to confirm it works in its current state for others and to working on sensor mapping.   It does have warnings in it for the fact is has some breaking changes and missing items.

**Reply by simieski** [reply]
Okay. I’ll give it a go.

BTW the only feature I’ve always felt I would like to add to this integration. Is a lock binary sensor. For people who want to use the lock state in automations but do not want to expose a lock entity for security reasons. Would this be straight forward to add? I used to have this with the bmw connected integration which is in core.

**Reply by cdnninja** [reply]
Easy to add but not understanding how it changes the security aspect?

**Reply by simieski** [reply]
Personally I’m very cautious of exposing a lock entity which can unlock my car. So I tend to disable the lock entity. In the past having a binary sensor for locked and unlocked enables me to disable the lock entity but still have automations based on the lock state.

**Reply by woody55uk** [reply]
I'd like to echo simieski's thanks for creating this integration. I have a Hyundai Kona EV in the UK and would also be happy to test anything. FYI I don't see the option to start my climate control in the entities available.

**Reply by 0rTiN** [reply]
> FYI I don't see the option to start my climate control in the entities available.

Its not an entitiy you have to look for, but a service. Just use "service: kia_uvo.start_climate" in a script or automation and your climate control should start. Wtih "service: kia_uvo.stop_climate" you are able to stop it.

**Reply by woody55uk** [reply]
OK - thanks. Found it now and works perfectly!

**Reply by naThnaabu** [reply]
Ioniq 5, happy to test if needed :)

**Reply by cdnninja** [reply]
Official integration isn't happening anymore.  We tried but received resistance merging it.

**Reply by naThnaabu** [reply]
oh no :( any exact reason why?

**Reply by cdnninja** [reply]
The core team wanted a trimmed version of this.  Take a look at the pull requests to the core version.  No real benefit of core vs this version here.
